src/main/java/frc/robot/subsystems/intakeLimelight/IntakeLimelight.java

```diff
@@ -0,0 +1,511 @@
+     *
+     * <p>This becomes a vector [x_norm, y_norm, z_norm] * d = [x_target, diff_height, z_target]
+     *
+     * <p>x_norm * d = x_target y_norm * d = diff_height z_target * d = z_target
+     */
+    double scaling = diff_height / y_norm;
+    double x_target = scaling * x_norm;
+    double z_target = scaling * z_norm;
+    double distance = Math.hypot(z_target, x_target);
+
+    // Convert camera coordinates to our conventions
+    Translation2d result = new Translation2d(distance, new Rotation2d(z_target, -x_target));
+
+    // For NT so this function doesn't need to be called multiple times
+    m_lastDistance = distance;
+    m_lastX = result.getX();
+    m_lastY = result.getY();
+
+    return result;
+  }
+
+  /**
+   * Get a 2d translation from the camera to the target, including normalization to handle the
+   * effects of angle to target. See the below discussion.
+   * https://www.chiefdelphi.com/t/what-does-limelight-skew-actually-measure/381167/7
+   *
+   * @return Translation2d from the camera to the target. This is camera relative, x is out of the
+   *     camera and positive y is to the left from the camera's point of view.
+   */
+  public Translation2d getTargetTranslation() {
+    return getTargetTranslation(kTargetHeight);
+  }
+
+  public class NoteDetection {
+    public double latencySec;
+
+    /** Distance from camera */
+    public double distanceMeters;
+
+    /** CCW Yaw Angle */
+    public double yawAngleDeg;
+
+    public NoteDetection(double latencySec, double distanceMeters, double yawAngleDeg) {
+      this.latencySec = latencySec;
+      this.distanceMeters = distanceMeters;
+      this.yawAngleDeg = yawAngleDeg;
+    }
+  }
+
+  /**
+   * Looks for a note
+   *
+   * @return If empty, no note detected. First value is note yaw in degrees (ccw), second value is
+   *     distance in meters.
+   */
+  public Optional<NoteDetection> getNotePos() {
+    // TODO filter low confidence detection in LL dashboard, hopefully
+    if (getPipeline() != pipeline.NOTE_PIPELINE) {
+      setPipeline(pipeline.NOTE_PIPELINE);
+    }
+
+    LimelightHelpers.LimelightResults llresults =
+        LimelightHelpers.getLatestResults(IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME);
+    LimelightTarget_Detector[] targets_Detector = llresults.targetingResults.targets_Detector;
+
+    if (targets_Detector.length == 0) {
+      System.out.println("Didn't see note");
+      return Optional.empty();
+    }
+
+    LimelightTarget_Detector lowestDetection = targets_Detector[0];
+    for (LimelightTarget_Detector detection : targets_Detector) {
+      if (detection.className == "note") {
+        if (detection.ty_pixels > lowestDetection.ty_pixels) { // lower in image = greater y value
+          lowestDetection = detection;
+        }
+      }
+    }
+
+    double angleLimelightToNoteDegrees =
+        IntakeLimelightConstants.INTAKE_LIMELIGHT_PITCH_DEGREES + lowestDetection.ty;
```

Comment:
The way we're using this below is that it's the angle relative to field horizontal, from the limelight position. So if the result (note relative to field horizontal) is 4 deg down and the limelight is 7 deg up. Then the ty will report 11. So I think we need `ty - INTAKE_LIMELIGHT_PITCH_DEGREES `

Reply:
Yup, you're right. Fixed